File: Manual/LightField/Add-in and Automation SDK/Samples/VisualBasic Add-Ins/Display Controls Sample/DisplayControlsWidgets.vb
'Enum of control types to be dynamically constructed
Public Enum ControlWidgetType
    SliderBar
    CheckBox
    EditBox
    ComboBox
    ReadOnlyEditBox
    Button
    Expander
    SubExpander
    SubEnd
End Enum

'ControlWidget class
Public Class ControlWidget
    Private name_ As String
    Private value_ As Object
    Private type_ As ControlWidgetType
    Private sliderMin_ As Double
    Private sliderMax_ As Double
    Private item_ As Object

    'The real physical control
    Public Property ControlItem As Object
        Get
            Return item_
        End Get
        Set(value As Object)
            item_ = value
        End Set
    End Property

    'Constructor non-slider
    Public Sub New(name As String, type As ControlWidgetType, value As Object)
        name_ = name
        type_ = type
        value_ = value
    End Sub

    'Constructor slider
    Public Sub New(name As String, type As ControlWidgetType, value As Object, min As Double, max As Double)
        name_ = name
        type_ = type
        value_ = value
        sliderMin_ = min
        sliderMax_ = max
    End Sub

    Public Function GetWidgetType() As ControlWidgetType
        Return type_
    End Function

    Public Function GetName() As String
        Return name_
    End Function

    Public Function GetValue() As Object
        Return value_
    End Function

    Public Function SliderMin() As Double
        Return sliderMin_
    End Function

    Public Function SliderMax() As Double
        Return sliderMax_
    End Function
End Class

File: Manual/LightField/Add-in and Automation SDK/Samples/CSharp Add-Ins/Display Controls Sample/DisplayControlsWidgets.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace LightFieldAddInSamples
{
    // Enum of Control Types To Be Dynamically Constructed
    public enum ControlWidgetType
    {
        SliderBar,
        CheckBox,
        EditBox,
        ComboBox,
        ReadOnlyEditBox,
        Button,
        Expander,
        SubExpander,
        SubEnd
    };
    // Control Widget class
    public class ControlWidget
    {
        string name_;
        object value_;
        ControlWidgetType type_;

        double sliderMin_;
        double sliderMax_;

        // The real physical control
        public object ControlItem { get; set; }

        // Constructor Non-Slider
        public ControlWidget(string name,
                                    ControlWidgetType type,
                                    object value)
        {
            name_ = name;
            type_ = type;
            value_ = value;
        }
        // Constructor Slider
        public ControlWidget(string name,
                             ControlWidgetType type,
                             object value,
                             double min,
                             double max)
        {
            name_ = name;
            type_ = type;
            value_ = value;
            sliderMin_ = min;
            sliderMax_ = max;
        }
        public ControlWidgetType GetWidgetType() { return type_; }
        public string GetName() { return name_; }
        public object GetValue() { return value_; }
        public double SliderMin() { return sliderMin_; }
        public double SliderMax() { return sliderMax_; }
    }
}
